Tests/CustomCode/ParseTests.cs

```diff
@@ -34,24 +35,47 @@ namespace UnitsNet.Tests.CustomCode
     [TestFixture]
     public class ParseTests
     {
-        /// <exception cref="UnitsNetException">Error parsing string.</exception>
-        [Test]
-        public void ParseWithUsEnglishCulture()
+        [TestCase("1km", Result=1000)]
+        [TestCase("1 km", Result = 1000)]
+        [TestCase("1e-3 km", Result = 1)]
+        [TestCase("5.5 m", Result = 5.5)]
+        [TestCase("500,005 m", Result = 500005)]
+        [TestCase(null, ExpectedExceptionName = "System.ArgumentNullException")]
+        [TestCase("1", ExpectedExceptionName = "System.ArgumentException")]
+        [TestCase("km", ExpectedExceptionName = "System.ArgumentException")]
+        [TestCase("1 kg", ExpectedExceptionName = "UnitsNet.UnitsNetException")]
+        public double ParseLengthToMetersUsEnglish(string s)
         {
             var usEnglish = CultureInfo.GetCultureInfo("en-US");
-            Assert.AreEqual(5.5, Length.Parse("5.5 m", usEnglish).Meters);
-            Assert.AreEqual(500005, Length.Parse("500,005 m", usEnglish).Meters);
+
+            return Length.Parse(s, usEnglish).Meters;
         }
 
         /// <exception cref="UnitsNetException">Error parsing string.</exception>
-        [Test]
-        public void ParseWithCultureUsingSpaceAsThousandSeparators()
+        [TestCase("5.5 m", Result = 5.5)]
+        [TestCase("500 005 m", Result = 500005)]
+        [TestCase("500.005.050,001 m", ExpectedExceptionName = "UnitsNet.UnitsNetException")] // quantity doesn't match number format
+        public double ParseWithCultureUsingSpaceAsThousandSeparators(string s)
         {
-            var numberFormat = (NumberFormatInfo) CultureInfo.InvariantCulture.NumberFormat.Clone();
+            var numberFormat = (NumberFormatInfo)CultureInfo.InvariantCulture.NumberFormat.Clone();
             numberFormat.NumberGroupSeparator = " ";
+            numberFormat.NumberDecimalSeparator = ".";
+
+            return Length.Parse(s, numberFormat).Meters;
+        }
+
+        /// <exception cref="UnitsNetException">Error parsing string.</exception>
+        [TestCase("5,5 m", Result = 5.5)]
+        [TestCase("500.005.050,001 m", Result = 500005050.001)]
+        [TestCase("5.5 m", Result = 55)] // dot is group separator not decimal
```

Comment:
This case is not intuitive to me. I thought groups were only for thousands, at least for invariant culture?